pkgs/development/python-modules/outlines-core/default.nix

```diff
@@ -0,0 +1,95 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+  pythonOlder,
+  cargo,
+  pkg-config,
+  rustPlatform,
+  rustc,
+  openssl,
+  setuptools-rust,
+  setuptools-scm,
+  interegular,
+  jsonschema,
+  datasets,
+  numpy,
+  pytestCheckHook,
+  pydantic,
+  scipy,
+  torch,
+  transformers,
+}:
+
+buildPythonPackage rec {
+  pname = "outlines-core";
+  version = "0.1.26";
+  pyproject = true;
+
+  disabled = pythonOlder "3.8";
+
+  src = fetchPypi {
+    inherit version;
+    pname = "outlines_core";
+    hash = "sha256-SBxDATQed8yPGDLWFnhK201GG0/sZYeOfA0sunFjoYk=";
+  };
+
+  cargoDeps = rustPlatform.importCargoLock {
+    lockFile = ./Cargo.lock;
+  };
+
+  postPatch = ''
+    cp --no-preserve=mode ${./Cargo.lock} Cargo.lock
+  '';
```

Comment:
Can we use cargoVendor instead?

Reply:
I can't find any reference to `cargoVendor` in nixpkgs? Did you mean `cargoLock`? If so, doesn't work because we are in `buildPythonPackage`.

Reply:
https://github.com/NixOS/nixpkgs/blob/e6914ee14cc9ef2221e9ac323c59cab1ae029684/pkgs/by-name/po/popsicle/package.nix#L25-L28

Reply:
But upstream does not ship a `Cargo.lock`, that's I had to add it, not because it has git dependencies.

Reply:
🤔 hmmm that might be a problem indeed